### ISSUE TITLE
`duckdb.from_arrow` fails with string_view data types

### ISSUE THREAD

**Issue by kylebarron** [duckdb/duckdb#13424]
### What happens?

When calling `from_arrow` on a pyarrow table with a string view column, import fails. 

From the Git blame it looks like support for this was implemented 6 months ago:
https://github.com/duckdb/duckdb/blob/4e3a192ce94a793510f11b598805f104d7531c15/src/function/table/arrow.cpp#L88-L89

But given that duckdb 1.0.0 was released in June, there might be a different bug here.

### To Reproduce

```py
import polars as pl
import duckdb
import pyarrow as pa

df = pl.DataFrame({"a": ["a", "b", "c"]})
table = pa.table(df)
print(table)
```
```
pyarrow.Table
a: string_view
----
a: [["a","b","c"]]
```

```py
duckdb.from_arrow(table)
```
raises

```
---------------------------------------------------------------------------
NotImplementedException                   Traceback (most recent call last)
File /Users/kyle/github/kylebarron/arro3/tests/core/test_ffi.py:1
----> 1 duckdb.from_arrow(table)

File ~/github/kylebarron/arro3/.venv/lib/python3.11/site-packages/duckdb/__init__.py:505, in from_arrow(arrow_object, **kwargs)
    503 else:
    504     conn = duckdb.connect(\":default:\")
--> 505 return conn.from_arrow(arrow_object, **kwargs)

NotImplementedException: Not implemented Error: Unsupported Internal Arrow Type vu
```

This is with polars 1.4.1, duckdb 1.0.0, pyarrow 17.0.0, Python 3.11.4.

### OS:

macOS arm Sonoma 14.5 (23F79)

### DuckDB Version:

1.0.0

### DuckDB Client:

Python

### Full Name:

Kyle Barron

### Affiliation:

Development Seed

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a stable release

### Did you include all relevant data sets for reproducing the issue?

Yes

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

**Comment by kylebarron** [comment]
Presumably the reason why no one has hit this before is because the _default_ way for polars to export to arrow is with `CompatLevel.oldest()`

```py
df = pl.DataFrame({"a": ["a", "b", "c"]})
df.to_arrow()
```
gives LargeString output:

```
pyarrow.Table
a: large_string
----
a: [["a","b","c"]]
```

and that loads in duckdb just fine.

Meanwhile if you pass `CompatLevel.newest()` to `DataFrame.to_arrow()`, polars exports to string view, and you get this issue:

```py
df = pl.DataFrame({"a": ["a", "b", "c"]})
df.to_arrow(compat_level=pl.CompatLevel.newest())
```

```
pyarrow.Table
a: string_view
----
a: [["a","b","c"]]
```

As of polars 1.3, Polars implements the [Arrow PyCapsule Interface](https://arrow.apache.org/docs/format/CDataInterface/PyCapsuleInterface.html), which is used when you pass a Polars DataFrame to the `pyarrow.table` constructor. This by default uses `CompatLevel.newest()` under the hood.

**Comment by Tishj** [comment]
That's a PyArrow error: `Not implemented Error: Unsupported Internal Arrow Type vu` is not raised by DuckDB

**Comment by Tishj** [comment]
This is fixed on `main`, I can reproduce it with v1.0.0, but with latest I can't anymore
